solr/solrj/src/test/org/apache/solr/client/solrj/SolrJMetricTestUtils.java

```diff
@@ -36,20 +36,17 @@ public static double getPrometheusMetricValue(CloudSolrClient solrClient, String
             SolrRequest.METHOD.GET,
             "/admin/metrics",
             SolrRequest.SolrRequestType.ADMIN,
-            new ModifiableSolrParams().set("wt", "prometheus"));
+            SolrParams.of("wt", "prometheus"));
     req.setResponseParser(new InputStreamResponseParser("prometheus"));
 
     NamedList<Object> resp = solrClient.request(req);
     try (InputStream in = (InputStream) resp.get("stream")) {
       String output = new String(in.readAllBytes(), StandardCharsets.UTF_8);
-      var line =
-          output
-              .lines()
-              .filter(l -> l.startsWith(metricName))
-              .findFirst()
-              .orElseThrow(() -> new AssertionError("Metric not found: " + metricName));
-
-      return Double.parseDouble(line.substring(line.lastIndexOf(" ") + 1).trim());
+      return output
+          .lines()
+          .filter(l -> l.startsWith(metricName))
+          .mapToDouble(s -> Double.parseDouble(s.substring(s.lastIndexOf(" ") + 1).trim()))
```

Comment:
The plus one results in a space that you then remove with a trim.  So can't you skip both and simplify?